In [4]:
import pandas as pd
from pathlib import Path

if not Path("olympics_1896_2004.csv").exists():
  !wget https://github.com/jonfernandes/pandas_essential/raw/main/olympics_1896_2004.csv

filename = "olympics_1896_2004.csv"
print("Installed all of the necessary files for this section ...")

Installed all of the necessary files for this section ...


In [10]:
df = pd.read_csv(filename, skiprows=5)
df.head()

Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0  1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens    Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   
2  1896  Athens  Athletics      Athletics       LANE, Francis  USA    Men   
3  1896  Athens  Athletics      Athletics    SZOKOLYI, Alajos  HUN    Men   
4  1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   

   Event Event Gender   Medal  Position  
0  100km            M    Gold         1  
1  100km            M  Silver         2  
2   100m            M  Bronze         3  
3   100m            M  Bronze         3  
4   100m            M    Gold         1

# 1. Фільтрація в Pandas
Звернення за індексом

In [6]:
# Отримати перший рядок за номером рядка
print(df.iloc[0])

Year                     1896
City                   Athens
Sport                 Cycling
Discipline      Cycling Track
Athlete Name    FLAMENG, Léon
NOC                       FRA
Gender                    Men
Event                   100km
Event Gender                M
Medal                    Gold
Position                    1
Name: 0, dtype: object


In [7]:
# Отримати останній рядок за номером рядка
print(df.iloc[-1])

Year                         2004
City                       Athens
Sport                     Fencing
Discipline                Fencing
Athlete Name    HEIDEMANN, Britta
NOC                           GER
Gender                      Women
Event                   épée team
Event Gender                    W
Medal                      Silver
Position                        2
Name: 27173, dtype: object


In [13]:
df.columns

Index(['Year', 'City', 'Sport', 'Discipline', 'Athlete Name', 'NOC', 'Gender',
       'Event', 'Event Gender', 'Medal', 'Position'],
      dtype='object')

In [14]:
# Використання loc для доступу до рядка за індексом та вибору конкретних колонок
print(df.loc[0, ['Athlete Name', 'Sport']])

Athlete Name    FLAMENG, Léon
Sport                 Cycling
Name: 0, dtype: object


In [25]:
print(df.loc[[0,3,5], 'Sport':'NOC'])

       Sport     Discipline      Athlete Name  NOC
0    Cycling  Cycling Track     FLAMENG, Léon  FRA
3  Athletics      Athletics  SZOKOLYI, Alajos  HUN
5  Athletics      Athletics    HOFMANN, Fritz  GER


In [9]:
# Використання iloc для вибору рядка за індексом та колонок за порядком
print(df.iloc[0, [4, 2]])  # Athlete Name and Sport

Athlete Name    FLAMENG, Léon
Sport                 Cycling
Name: 0, dtype: object


Використання [] для фільтрації вибірки

Фільтрація даних на основі умов:

In [26]:
# Фільтрація атлетів, що отримали золоту медаль - loc method
gold_medals = df.loc[df['Medal'] == 'Gold']
print(gold_medals.head())

    Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0   1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
4   1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   
6   1896  Athens   Aquatics       Swimming       HAJOS, Alfred  HUN    Men   
9   1896  Athens   Aquatics       Swimming  MALOKINIS, Ioannis  GRE    Men   
12  1896  Athens    Cycling  Cycling Track        MASSON, Paul  FRA    Men   

                         Event Event Gender Medal  Position  Total_Medals  
0                        100km            M  Gold         1          1238  
4                         100m            M  Gold         1          4020  
6               100m freestyle            M  Gold         1          1026  
9   100m freestyle for sailors            M  Gold         1           138  
12                        10km            M  Gold         1          1238  


In [15]:
# Фільтрація атлетів, що отримали золоту медаль - simple syntax
gold_medals = df[df['Medal'] == 'Gold']
print(gold_medals.head())

    Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0   1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
4   1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   
6   1896  Athens   Aquatics       Swimming       HAJOS, Alfred  HUN    Men   
9   1896  Athens   Aquatics       Swimming  MALOKINIS, Ioannis  GRE    Men   
12  1896  Athens    Cycling  Cycling Track        MASSON, Paul  FRA    Men   

                         Event Event Gender Medal  Position  
0                        100km            M  Gold         1  
4                         100m            M  Gold         1  
6               100m freestyle            M  Gold         1  
9   100m freestyle for sailors            M  Gold         1  
12                        10km            M  Gold         1  


Фільтрація за кількома умовами (використовуйте оператори & (і) або | (або):)

In [16]:
# Фільтрація атлетів, які отримали золоту медаль у чоловічих змаганнях
gold_men = df[(df['Medal'] == 'Gold') & (df['Gender'] == 'Men')]
print(gold_men.head())

    Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0   1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
4   1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   
6   1896  Athens   Aquatics       Swimming       HAJOS, Alfred  HUN    Men   
9   1896  Athens   Aquatics       Swimming  MALOKINIS, Ioannis  GRE    Men   
12  1896  Athens    Cycling  Cycling Track        MASSON, Paul  FRA    Men   

                         Event Event Gender Medal  Position  
0                        100km            M  Gold         1  
4                         100m            M  Gold         1  
6               100m freestyle            M  Gold         1  
9   100m freestyle for sailors            M  Gold         1  
12                        10km            M  Gold         1  


In [27]:
# Використання методу isin для перевірки умови на відповідність декільком значенням
gold_men = df[(df['Medal'].isin(['Gold','Silver'])) & (df['Gender'] == 'Men')]
print(gold_men.head())

   Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0  1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens    Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   
4  1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   
5  1896  Athens  Athletics      Athletics      HOFMANN, Fritz  GER    Men   
6  1896  Athens   Aquatics       Swimming       HAJOS, Alfred  HUN    Men   

            Event Event Gender   Medal  Position  Total_Medals  
0           100km            M    Gold         1          1238  
1           100km            M  Silver         2           138  
4            100m            M    Gold         1          4020  
5            100m            M  Silver         2          1110  
6  100m freestyle            M    Gold         1          1026  


In [28]:
# Використання методу isin для перевірки умови на НЕвідповідність декільком значенням
gold_men = df[(~df['Medal'].isin(['Gold','Silver'])) & (df['Gender'] == 'Men')]
print(gold_men.head())

    Year    City      Sport     Discipline           Athlete Name  NOC Gender  \
2   1896  Athens  Athletics      Athletics          LANE, Francis  USA    Men   
3   1896  Athens  Athletics      Athletics       SZOKOLYI, Alajos  HUN    Men   
8   1896  Athens   Aquatics       Swimming      DRIVAS, Dimitrios  GRE    Men   
11  1896  Athens    Cycling  Cycling Track          SCHMAL, Adolf  AUT    Men   
18  1896  Athens   Aquatics       Swimming  CHOROPHAS, Efstathios  GRE    Men   

                         Event Event Gender   Medal  Position  Total_Medals  
2                         100m            M  Bronze         3          4020  
3                         100m            M  Bronze         3          1026  
8   100m freestyle for sailors            M  Bronze         3           138  
11                        10km            M  Bronze         3           143  
18             1200m freestyle            M  Bronze         3           138  


In [29]:
# перевірка порожніх та непорожніх значень
df[df['Gender'].isna()].head()

Empty DataFrame
Columns: [Year, City, Sport, Discipline, Athlete Name, NOC, Gender, Event, Event Gender, Medal, Position, Total_Medals]
Index: []

In [30]:
df[df['Gender'].notna()].head()

Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0  1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens    Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   
2  1896  Athens  Athletics      Athletics       LANE, Francis  USA    Men   
3  1896  Athens  Athletics      Athletics    SZOKOLYI, Alajos  HUN    Men   
4  1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   

   Event Event Gender   Medal  Position  Total_Medals  
0  100km            M    Gold         1          1238  
1  100km            M  Silver         2           138  
2   100m            M  Bronze         3          4020  
3   100m            M  Bronze         3          1026  
4   100m            M    Gold         1          4020

Створення мультиіндексів

In [31]:
indexed_df = df.set_index(['Sport', 'Discipline', 'NOC'])
indexed_df.head()

Year    City        Athlete Name Gender  Event  \
Sport     Discipline    NOC                                                   
Cycling   Cycling Track FRA  1896  Athens       FLAMENG, Léon    Men  100km   
                        GRE  1896  Athens  KOLETTIS, Georgios    Men  100km   
Athletics Athletics     USA  1896  Athens       LANE, Francis    Men   100m   
                        HUN  1896  Athens    SZOKOLYI, Alajos    Men   100m   
                        USA  1896  Athens       BURKE, Thomas    Men   100m   

                            Event Gender   Medal  Position  Total_Medals  
Sport     Discipline    NOC                                               
Cycling   Cycling Track FRA            M    Gold         1          1238  
                        GRE            M  Silver         2           138  
Athletics Athletics     USA            M  Bronze         3          4020  
                        HUN            M  Bronze         3          1026  
                        USA            M    Gold         1          4020

In [33]:
# Одержання даних за мультиіндексом через метод loc
indexed_df.loc['Cycling']

Year    City            Athlete Name Gender        Event  \
Discipline    NOC                                                             
Cycling Track FRA  1896  Athens           FLAMENG, Léon    Men        100km   
              GRE  1896  Athens      KOLETTIS, Georgios    Men        100km   
              AUT  1896  Athens           SCHMAL, Adolf    Men         10km   
              FRA  1896  Athens            MASSON, Paul    Men         10km   
              FRA  1896  Athens           FLAMENG, Léon    Men         10km   
...                 ...     ...                     ...    ...          ...   
              RUS  2004  Athens        SLYUSAREVA, Olga  Women  points race   
              MEX  2004  Athens  GUERRERO MENDEZ, Belem  Women  points race   
              AUS  2004  Athens            MEARES, Anna  Women       sprint   
              CAN  2004  Athens       MUENZER, Lori-Ann  Women       sprint   
              RUS  2004  Athens       ABASSOVA, Tamilla  Women       sprint   

                  Event Gender   Medal  Position  Total_Medals  
Discipline    NOC                                               
Cycling Track FRA            M    Gold         1          1238  
              GRE            M  Silver         2           138  
              AUT            M  Bronze         3           143  
              FRA            M    Gold         1          1238  
              FRA            M  Silver         2          1238  
...                        ...     ...       ...           ...  
              RUS            W    Gold         1           495  
              MEX            W  Silver         2            76  
              AUS            W  Bronze         3           926  
              CAN            W    Gold         1           558  
              RUS            W  Silver         2           495  

[954 rows x 9 columns]

In [37]:
indexed_df.loc['Cycling', 'Cycling Track']

<ipython-input-37-e1f40f99f21d>:1: PerformanceWarning: indexing past lexsort depth may impact performance.
  indexed_df.loc['Cycling', 'Cycling Track']


Year    City            Athlete Name Gender        Event Event Gender  \
NOC                                                                          
FRA  1896  Athens           FLAMENG, Léon    Men        100km            M   
GRE  1896  Athens      KOLETTIS, Georgios    Men        100km            M   
AUT  1896  Athens           SCHMAL, Adolf    Men         10km            M   
FRA  1896  Athens            MASSON, Paul    Men         10km            M   
FRA  1896  Athens           FLAMENG, Léon    Men         10km            M   
..    ...     ...                     ...    ...          ...          ...   
RUS  2004  Athens        SLYUSAREVA, Olga  Women  points race            W   
MEX  2004  Athens  GUERRERO MENDEZ, Belem  Women  points race            W   
AUS  2004  Athens            MEARES, Anna  Women       sprint            W   
CAN  2004  Athens       MUENZER, Lori-Ann  Women       sprint            W   
RUS  2004  Athens       ABASSOVA, Tamilla  Women       sprint            W   

      Medal  Position  Total_Medals  
NOC                                  
FRA    Gold         1          1238  
GRE  Silver         2           138  
AUT  Bronze         3           143  
FRA    Gold         1          1238  
FRA  Silver         2          1238  
..      ...       ...           ...  
RUS    Gold         1           495  
MEX  Silver         2            76  
AUS  Bronze         3           926  
CAN    Gold         1           558  
RUS  Silver         2           495  

[644 rows x 9 columns]

In [38]:
indexed_df.loc['Cycling', 'Cycling Track', 'FRA']

<ipython-input-38-83dae29bd21c>:1: PerformanceWarning: indexing past lexsort depth may impact performance.
  indexed_df.loc['Cycling', 'Cycling Track', 'FRA']


Year    City        Athlete Name Gender  \
Sport   Discipline    NOC                                            
Cycling Cycling Track FRA  1896  Athens       FLAMENG, Léon    Men   
                      FRA  1896  Athens        MASSON, Paul    Men   
                      FRA  1896  Athens       FLAMENG, Léon    Men   
                      FRA  1896  Athens        MASSON, Paul    Men   
                      FRA  1896  Athens       FLAMENG, Léon    Men   
...                         ...     ...                 ...    ...   
                      FRA  2000  Sydney  BALLANGER, Felicia  Women   
                      FRA  2004  Athens    TOURNANT, Arnaud    Men   
                      FRA  2004  Athens   BOURGAIN, Mickael    Men   
                      FRA  2004  Athens       GANE, Laurent    Men   
                      FRA  2004  Athens    TOURNANT, Arnaud    Men   

                                     Event Event Gender   Medal  Position  \
Sport   Discipline    NOC                                                   
Cycling Cycling Track FRA            100km            M    Gold         1   
                      FRA             10km            M    Gold         1   
                      FRA             10km            M  Silver         2   
                      FRA   1km time trial            M    Gold         1   
                      FRA  Sprint indivual            M  Bronze         3   
...                                    ...          ...     ...       ...   
                      FRA           sprint            W    Gold         1   
                      FRA   1km time trial            M  Silver         2   
                      FRA      Team Sprint            M  Bronze         3   
                      FRA      Team Sprint            M  Bronze         3   
                      FRA      Team Sprint            M  Bronze         3   

                           Total_Medals  
Sport   Discipline    NOC                
Cycling Cycling Track FRA          1238  
                      FRA          1238  
                      FRA          1238  
                      FRA          1238  
                      FRA          1238  
...                                 ...  
                      FRA          1238  
                      FRA          1238  
                      FRA          1238  
                      FRA          1238  
                      FRA          1238  

[85 rows x 9 columns]

In [39]:
indexed_df.loc['Cycling', 'FRA']

<ipython-input-39-55c89e2c76b9>:1: PerformanceWarning: indexing past lexsort depth may impact performance.
  indexed_df.loc['Cycling', 'FRA']


KeyError: 'FRA'

In [43]:
indexed_df.loc['Cycling', :, 'FRA']

Year    City       Athlete Name Gender            Event  \
Discipline                                                               
Cycling Track  1896  Athens      FLAMENG, Léon    Men            100km   
Cycling Track  1896  Athens       MASSON, Paul    Men             10km   
Cycling Track  1896  Athens      FLAMENG, Léon    Men             10km   
Cycling Track  1896  Athens       MASSON, Paul    Men   1km time trial   
Cycling Track  1896  Athens      FLAMENG, Léon    Men  Sprint indivual   
...             ...     ...                ...    ...              ...   
Cycling Track  2004  Athens   TOURNANT, Arnaud    Men   1km time trial   
Cycling Track  2004  Athens  BOURGAIN, Mickael    Men      Team Sprint   
Cycling Track  2004  Athens      GANE, Laurent    Men      Team Sprint   
Cycling Track  2004  Athens   TOURNANT, Arnaud    Men      Team Sprint   
Mountain Bike  2004  Athens    ABSALON, Julien    Men    cross-country   

              Event Gender   Medal  Position  Total_Medals  
Discipline                                                  
Cycling Track            M    Gold         1          1238  
Cycling Track            M    Gold         1          1238  
Cycling Track            M  Silver         2          1238  
Cycling Track            M    Gold         1          1238  
Cycling Track            M  Bronze         3          1238  
...                    ...     ...       ...           ...  
Cycling Track            M  Silver         2          1238  
Cycling Track            M  Bronze         3          1238  
Cycling Track            M  Bronze         3          1238  
Cycling Track            M  Bronze         3          1238  
Mountain Bike            M    Gold         1          1238  

[122 rows x 9 columns]

# 2. Групування, агрегація та сортування в Pandas
Застосування функцій агрегації на Series

In [44]:
# Підрахунок кількості медалей за дисципліною
medals_by_discipline = df['Discipline'].value_counts()
print(medals_by_discipline.head())

Discipline
Athletics      3271
Rowing         2379
Swimming       2239
Artistic G.    2031
Fencing        1485
Name: count, dtype: int64


Групування за значеннями полів

In [19]:
# Групування медалей за країною (NOC)
medals_by_noc = df.groupby('NOC')['Medal'].count()
print(medals_by_noc)

NOC
AHO      1
ALG     12
ANZ     29
ARG    188
ARM      3
      ... 
VIE      1
YUG    435
ZAM      2
ZIM     19
ZZX     48
Name: Medal, Length: 132, dtype: int64


In [45]:
medals_by_noc = medals_by_noc.sort_values(ascending=False)
medals_by_noc.head()

NOC
USA    4020
URS    2049
GBR    1517
FRA    1238
ITA    1186
Name: Medal, dtype: int64

In [48]:
type(medals_by_noc)

pandas.core.series.Series

In [49]:
noc_medals_df = pd.DataFrame(medals_by_noc)
type(noc_medals_df)

pandas.core.frame.DataFrame

In [51]:
noc_medals_df.head()

Medal
NOC       
AHO      1
ALG     12
ANZ     29
ARG    188
ARM      3

In [65]:
noc_medals_df.columns

Index(['Medal'], dtype='object')

In [53]:
# сума всіх значень у серії чи стовпці
noc_medals_df['Medal'].sum()

27174

In [54]:
# середнє арифметичне значення
noc_medals_df['Medal'].mean()

205.86363636363637

In [55]:
# медіана
noc_medals_df['Medal'].median()

19.0

In [56]:
# Мінімальне значення
noc_medals_df['Medal'].min()

1

In [63]:
# Індекс мінімального значення
noc_medals_df['Medal'].idxmin()

'AHO'

In [66]:
noc_medals_df.loc['AHO']

Medal    1
Name: AHO, dtype: int64

In [57]:
# Максимальне значення
noc_medals_df['Medal'].max()

4020

In [67]:
# Індекс максимального значення
noc_medals_df['Medal'].idxmax()

'USA'

In [69]:
noc_medals_df.loc['USA']

Medal    4020
Name: USA, dtype: int64

In [70]:
noc_medals_df.loc[noc_medals_df['Medal'].idxmax()]

Medal    4020
Name: USA, dtype: int64

In [58]:
# стандартне відхилення
noc_medals_df['Medal'].std()

475.53598161719594

In [59]:
# знаходження дисперсії
noc_medals_df['Medal'].var()

226134.4698126301

In [60]:
# кількість непустих значень
noc_medals_df['Medal'].count()

132

In [61]:
# Кількість унікальних значень
noc_medals_df['Medal'].nunique()

75

In [62]:
# Знаходження квантиля
noc_medals_df['Medal'].quantile(0.75)

178.25

In [72]:
# Стандартна помилка середнього
noc_medals_df['Medal'].sem()

41.39009450125664

Застосування функцій агрегації на згрупованих DataFrames

In [77]:
# Групування медалей за країною (NOC)
medals_by_noc = df.groupby('NOC', as_index=False, sort=False)['Medal'].count()
print(medals_by_noc)

     NOC  Medal
0    FRA   1238
1    GRE    138
2    USA   4020
3    HUN   1026
4    GER   1110
..   ...    ...
127  KUW      1
128  ERI      1
129  SCG     14
130  UAE      1
131  PAR     17

[132 rows x 2 columns]


In [86]:
# Підрахунок мінімального і максимального місця на пʼєдесталі для кожної країни
agg_data = df.groupby('NOC')['Position'].agg(['min', 'max'])
print(agg_data)

     min  max
NOC          
AHO    2    2
ALG    1    3
ANZ    1    3
ARG    1    3
ARM    1    3
..   ...  ...
VIE    2    2
YUG    1    3
ZAM    2    3
ZIM    1    3
ZZX    1    3

[132 rows x 2 columns]


In [20]:
# Підрахунок кількості медалей та визначення унікальних видів спорту для кожної країни
agg_data = df.groupby('NOC').agg({
    'Medal': 'count',
    'Sport': 'nunique'
})
print(agg_data)

     Medal  Sport
NOC              
AHO      1      1
ALG     12      2
ANZ     29      5
ARG    188     15
ARM      3      2
..     ...    ...
VIE      1      1
YUG    435     14
ZAM      2      2
ZIM     19      2
ZZX     48      6

[132 rows x 2 columns]


Застосування методу transform()

In [21]:
# Додавання колонки з кількістю медалей для кожної країни (групування і трансформація)
df['Total_Medals'] = df.groupby('NOC')['Medal'].transform('count')
print(df.head())

   Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0  1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens    Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   
2  1896  Athens  Athletics      Athletics       LANE, Francis  USA    Men   
3  1896  Athens  Athletics      Athletics    SZOKOLYI, Alajos  HUN    Men   
4  1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   

   Event Event Gender   Medal  Position  Total_Medals  
0  100km            M    Gold         1          1238  
1  100km            M  Silver         2           138  
2   100m            M  Bronze         3          4020  
3   100m            M  Bronze         3          1026  
4   100m            M    Gold         1          4020  


Сортування в Pandas

In [23]:
# Сортування атлетів за роком змагань та ім'ям
sorted_df = df.sort_values(by=['Year', 'Athlete Name'])
print(sorted_df.head())

     Year    City       Sport    Discipline              Athlete Name  NOC  \
20   1896  Athens    Aquatics      Swimming          ANDREOU, Joannis  GRE   
103  1896  Athens  Gymnastics   Artistic G.  ANDRIAKOPOULOS, Nicolaos  GRE   
141  1896  Athens  Gymnastics   Artistic G.  ANDRIAKOPOULOS, Nicolaos  GRE   
142  1896  Athens  Gymnastics   Artistic G.    ATHANASOPOULOS, Spyros  GRE   
79   1896  Athens     Cycling  Cycling Road            BATTEL, Edward  GBR   

    Gender                 Event Event Gender   Medal  Position  Total_Medals  
20     Men       1200m freestyle            M  Silver         2           138  
103    Men         rope climbing            M    Gold         1           138  
141    Men   team, parallel bars            M  Silver         2           138  
142    Men   team, parallel bars            M  Silver         2           138  
79     Men  individual road race            M  Bronze         3          1517  


# Спробуйте вирішити ці завдання? Чи допоможе вам аналіз даних чи пошук у Google?

Answer the following questions. Indicate the Pandas command where relevant.

What is the time range covered in this dataset?

The Olympics take place every 4 years. Why are there missing years?

What are the types of medals awarded?

Across all of the Olympic Games, how many Gold, Silver and Bronze medals have there been?

Why are there not an equal number of Gold, Silver and Bronze medals?

There are more Gold medals than Silver, and more Silver than Bronze. Why might that be?

What are the different NOCs (National Olympic Committees)
What does the NOC 'ZZX' represent?

In [87]:
import pandas as pd
from pathlib import Path

if not Path("olympics_1896_2004.csv").exists():
  !wget https://github.com/jonfernandes/pandas_essential/raw/main/olympics_1896_2004.csv

filename = "olympics_1896_2004.csv"
print("Installed all of the necessary files for this section ...")

Installed all of the necessary files for this section ...


1. Який період даних охоплено даних набором даних?

In [88]:
df.Year.unique()

array([1896, 1900, 1904, 1908, 1912, 1920, 1924, 1928, 1932, 1936, 1948,
       1952, 1956, 1960, 1964, 1968, 1972, 1976, 1980, 1984, 1988, 1992,
       1996, 2000, 2004])

2. Олімпіада проводиться один раз на 4 роки. Чому є пропуски в даних?

Перша світова війна: 1914-1918

Друга світова війна: 1939-1945

3. Якими типами медалей нагороджували спортсменів?

In [89]:
df.Medal.unique()

array(['Gold', 'Silver', 'Bronze'], dtype=object)

4. За весь час проведення Олімпіади скільки було виграно медалей кожного виду?

In [91]:
df.Medal.value_counts()

Medal
Gold      9181
Silver    9014
Bronze    8979
Name: count, dtype: int64

5. Чому кількість золотих, срібних і бронзових медалей не однакова?

- нічия
- командні види спорту
- дискваліфікації

6. Золотих медалей більше ніж срібних, а срібних більше ніж бронзових. Чому?

In [94]:
 df.head(10)

Year    City      Sport     Discipline        Athlete Name  NOC Gender  \
0  1896  Athens    Cycling  Cycling Track       FLAMENG, Léon  FRA    Men   
1  1896  Athens    Cycling  Cycling Track  KOLETTIS, Georgios  GRE    Men   
2  1896  Athens  Athletics      Athletics       LANE, Francis  USA    Men   
3  1896  Athens  Athletics      Athletics    SZOKOLYI, Alajos  HUN    Men   
4  1896  Athens  Athletics      Athletics       BURKE, Thomas  USA    Men   
5  1896  Athens  Athletics      Athletics      HOFMANN, Fritz  GER    Men   
6  1896  Athens   Aquatics       Swimming       HAJOS, Alfred  HUN    Men   
7  1896  Athens   Aquatics       Swimming    HERSCHMANN, Otto  AUT    Men   
8  1896  Athens   Aquatics       Swimming   DRIVAS, Dimitrios  GRE    Men   
9  1896  Athens   Aquatics       Swimming  MALOKINIS, Ioannis  GRE    Men   

                        Event Event Gender   Medal  Position  Total_Medals  
0                       100km            M    Gold         1          1238  
1                       100km            M  Silver         2           138  
2                        100m            M  Bronze         3          4020  
3                        100m            M  Bronze         3          1026  
4                        100m            M    Gold         1          4020  
5                        100m            M  Silver         2          1110  
6              100m freestyle            M    Gold         1          1026  
7              100m freestyle            M  Silver         2           143  
8  100m freestyle for sailors            M  Bronze         3           138  
9  100m freestyle for sailors            M    Gold         1           138

Чоловіки 100 км на велосипеді

https://en.wikipedia.org/wiki/Cycling_at_the_1896_Summer_Olympics_%E2%80%93_Men%27s_100_kilometres
https://en.wikipedia.org/wiki/Aristidis_Konstantinidis

Чоловіки 100 м плавання вільним стилем (1896)

https://en.wikipedia.org/wiki/Swimming_at_the_1896_Summer_Olympics_%E2%80%93_Men%27s_100_metre_freestyle

7. Які Олімпійські комітети були представлені на Олімпіадах?

In [92]:
df.NOC.unique()

array(['FRA', 'GRE', 'USA', 'HUN', 'GER', 'AUT', 'GBR', 'AUS', 'DEN',
       'ZZX', 'SUI', 'IND', 'NED', 'CAN', 'NOR', 'BEL', 'ESP', 'BOH',
       'ITA', 'SWE', 'CUB', 'RU1', 'FIN', 'RSA', 'ANZ', 'LUX', 'EST',
       'BRA', 'JPN', 'TCH', 'NZL', 'ARG', 'HAI', 'POL', 'URU', 'YUG',
       'ROU', 'POR', 'EGY', 'PHI', 'IRL', 'CHI', 'MEX', 'LAT', 'TUR',
       'KOR', 'PAN', 'JAM', 'SRI', 'PER', 'PUR', 'IRI', 'TRI', 'URS',
       'LIB', 'BUL', 'VEN', 'EUA', 'PAK', 'ISL', 'BAH', 'BWI', 'GHA',
       'IRQ', 'SIN', 'TPE', 'ETH', 'MAR', 'TUN', 'NGR', 'KEN', 'FRG',
       'MGL', 'GDR', 'UGA', 'CMR', 'PRK', 'COL', 'NIG', 'BER', 'THA',
       'TAN', 'GUY', 'ZIM', 'ZAM', 'CHN', 'CIV', 'DOM', 'ALG', 'SYR',
       'SUR', 'CRC', 'SEN', 'AHO', 'DJI', 'ISV', 'INA', 'EUN', 'NAM',
       'IOP', 'QAT', 'ISR', 'LTU', 'CRO', 'SLO', 'MAS', 'RUS', 'UKR',
       'TGA', 'ARM', 'BLR', 'MDA', 'ECU', 'KAZ', 'AZE', 'BDI', 'SVK',
       'CZE', 'UZB', 'GEO', 'MOZ', 'HKG', 'KGZ', 'BAR', 'KSA', 'VIE',
       'MKD', 'KUW',

8. З якої країни олімпійський комітет ZZX?

https://en.wikipedia.org/wiki/Mixed_teams_at_the_Olympic